### ISSUE TITLE
Feature Request: Add Design tool mode to allow digging/channeling a line between multiple Z levels

### ISSUE THREAD

**Issue by Dragoon209** [DFHack/dfhack#3371]
When using line mode across Z levels, it seems to function just like normal rectangle mode.  

My expectation would be that the line mode would work the same as it does on a single  Z level, connecting the 2 dots together.  

For a use case, I was attempting to channel a line across multiple Z levels for a minecart track.  I would love to be able to designate a across layers and have the tool place the mine and channel designations where needed to connect.  Even just placing the mining designations and then placing the channel marks at the end  of each level's section would be a huge help.

Thanks!

**Comment by myk002** [comment]
This is a very interesting idea. We might want to differentiate between "line" mode, where the same shape is duplicated on every level, and "tunnel" mode, which would be a single 3D shape drawn through the z-levels.

@TaxiService @johncosker thoughts?

**Comment by myk002** [comment]
similar question for other 3D shapes like "domes" and "archways"

**Comment by Dragoon209** [comment]
Ooh, domes and archways are very cool, didn't even thank of that.  That's awesome

**Comment by johncosker** [comment]
Yeah, right now the way it works with Z-levels across all shapes is to just apply the X/Y shape to each Z level.

I did have a 'TODO' plan for having 3d shapes but haven't gotten to it yet, it'd be a pretty significant refactor of how the shape code works right now, not that it's impossible or anything, though.

**Comment by lethosor** [comment]
The main difficulty I see here is that you may not want to dig in the same way on every tile. e.g. to make a multi-z-level line passable, you would need ramps or stairs in some places, while a line on the same level can be made passible just with ordinary digging.
